test/evp_test.c

```diff
@@ -1597,27 +1522,24 @@ static int pderive_test_parse(struct evp_test *t,
     return 0;
 }
 
-static int pderive_test_run(struct evp_test *t)
+static int pderive_test_run(EVP_TEST *t)
 {
-    struct pkey_data *kdata = t->data;
+    PKEY_DATA *kdata = t->data;
     unsigned char *out = NULL;
     size_t out_len;
     const char *err = "INTERNAL_ERROR";
```

Comment:
Maybe you should convert this function to assign `t->err` directly too, like you did higher up?